lms/product/canvas/_plugin/misc.py

```diff
@@ -136,13 +136,18 @@ def get_deep_linked_assignment_configuration(self, request) -> dict:
         return params
 
     def get_deeplinking_launch_url(self, request, assignment_configuration: dict):
-        # In canvas we point to our generic launch URL
-        # and we encode the assignment configuration as query parameters.
-        return (
-            urlparse(request.route_url("lti_launches"))
+        route_name = "lti_launches"
```

Comment:
This is a bit trickier because we have now two scenarios:

- Two endpoints, the deep linking launch returns our launch URL
- One endpoint, just use that one URL.

The complexity here is knowing in which case we are in. This particular view is *not* a launch so that information needs to be passed along.